### ISSUE TITLE
webserver add role to an existing user -> KeyError: 'userinfoedit'

### ISSUE THREAD

**Issue by raphaelauv** [apache/airflow#27479]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

airflow 2.4.1

a existing user only have the role Viewer
I add with the UI the role Admin
click on button save

then error ->

```log

[03/Nov/2022:01:28:08 +0000] "POST /XXXXXXXX/users/edit/2 HTTP/1.1" 302 307 "https://XXXXXXXXXXXXX.net/XXXXXXXX/users/edit/2" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:106.0) Gecko/20100101 Firefox/106.0"
[2022-11-03T01:28:09.014+0000] {app.py:1742} ERROR - Exception on /users/show/1 [GET]
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/airflow/.local/lib/python3.7/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/airflow/.local/lib/python3.7/site-packages/flask_appbuilder/security/decorators.py", line 133, in wraps
    return f(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/www/fab_security/views.py", line 222, in show
    widgets['show'].template_args['actions'].pop('userinfoedit')
KeyError: 'userinfoedit'
```

### Operating System

Ubuntu 20.04.4 LTS

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

1.7.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by tirkarthi** [comment]
Looks like a duplicate of https://github.com/apache/airflow/issues/22035

**Comment by uranusjr** [comment]
Unfortunately #22035 was closed due to inactivity, so this is still an issue.

**Comment by zaibacu** [comment]
Encountered this issue with Airflow version: 2.4.3 while chaning password.

Why it keeps getting closed while being totally valid issue? Collegue is unable to login and use Airflow because of that.

**Comment by potiuk** [comment]
BTW. This is how Open Source software development works. Nothing special there. But it's very diferent from "helpdesk" when you raise an issue and expect it to be handled magically. It might or might not happen, but by raisign an issue you also take a responsibility to watch over it.

**Comment by potiuk** [comment]
(And BTW. I reopened the issue - please subscribe to it so that it won't happen again that it gets closed),

**Comment by chyok** [comment]
> Encountered this issue with Airflow version: 2.4.3 while chaning password.
> 
> Why it keeps getting closed while being totally valid issue? Collegue is unable to login and use Airflow because of that.

@zaibacu #27537 This PR fixed the issue, so this issue has been closed. According to the release notes, this PR was first included in Airflow 2.5.0. Please try the versions 2.5.0 and later.
